src/test/java/htsjdk/tribble/AbstractFeatureReaderTest.java

```diff
@@ -103,10 +123,142 @@ public void testBlockCompressionExtensionFile(final String testString, final boo
         };
     }
 
-    @Test(enabled = true, dataProvider = "blockCompressedExtensionExtensionURIStrings")
+    @Test(dataProvider = "blockCompressedExtensionExtensionURIStrings")
     public void testBlockCompressionExtension(final String testURIString, final boolean expected) throws URISyntaxException {
         URI testURI = URI.create(testURIString);
         Assert.assertEquals(AbstractFeatureReader.hasBlockCompressedExtension(testURI), expected);
     }
 
+
+    @DataProvider(name = "vcfFileAndWrapperCombinations")
+    private static Object[][] vcfFileAndWrapperCombinations(){
+        return new Object[][] {
+                {VCF, VCF_INDEX, null, null},
+                {MANGLED_VCF, MANGLED_VCF_INDEX, WRAPPER, WRAPPER},
+                {VCF, MANGLED_VCF_INDEX, null, WRAPPER},
+                {MANGLED_VCF, VCF_INDEX, WRAPPER, null},
+                {MANGLED_VCF_TABIX, MANGLED_VCF_TABIX_INDEX, WRAPPER, WRAPPER},
+                {VCF_TABIX, MANGLED_VCF_TABIX_INDEX, null, WRAPPER},
+                {MANGLED_VCF_TABIX, VCF_TABIX_INDEX, WRAPPER, null},
```

Comment:
Have you confirmed that all test cases that take a wrapper actually throw if the wrapper is not provided?